src/parser/shared/modules/MitigationCheck.js

```diff
@@ -19,47 +20,77 @@ class MitigationCheck extends Analyzer {
     enemies: Enemies,
   };
 
-  checks = [];
-  buffCheck = [];
-  debuffCheck = [];
   checksPassedMap = new Map();
   checksFailedMap = new Map();
 
+  checksPhysical = [];
+  checksMagical = [];
+
+  buffCheckPhysical = [];
+  buffCheckMagical = [];
+  buffCheckPhysAndMag = [];
+
+  debuffCheckPhysical = [];
+  debuffCheckMagical = [];
+  debuffCheckPhysAndMag = [];
+
   constructor(...args) {
     super(...args);
     if (this.owner.boss == null) {
       this.active = false;
       return;
     }
     const boss = findByBossId(this.owner.boss.id);
-    if (boss.fight.softMitigationChecks) {
-      this.checks = boss.fight.softMitigationChecks;
-      if (this.checks === undefined) {
-        this.checks = [];
+    if (boss.fight.softMitigationChecks.physical && boss.fight.softMitigationChecks.magical) {
+      this.checksPhysical = boss.fight.softMitigationChecks.physical;
+      this.checksMagical = boss.fight.softMitigationChecks.magical;
+      if (this.checksPhysical === undefined || this.checksMagical === undefined) {
+        this.checksPhysical = [];
+        this.checksMagical = [];
       }
     } else {
-      this.checks = [];
-      this.buffCheck = [];
-      this.debuffCheck = [];
+      this.checksPhysical = [];
+      this.checksMagical = [];
+
+      this.buffCheckPhysical = [];
+      this.buffCheckMagical = [];
+      this.buffCheckPhysAndMag = [];
+
+      this.debuffCheckPhysical = [];
+      this.debuffCheckMagical = [];
+      this.debuffCheckPhysAndMag = [];
     }
-    this.checks.forEach((e) => {
+    [...this.checksPhysical, ...this.checksMagical].forEach((e) => {
       this.checksPassedMap.set(e, 0);
       this.checksFailedMap.set(e, 0);
     });
   }
 
   on_toPlayer_damage(event) {
     const spell = event.ability.guid;
-    if (this.checks.includes(spell) && !event.tick) {
-      debug && console.log(this.buffCheck);
-      debug && console.log(this.debuffCheck);
-      if (this.buffCheck.some((e) => this.selectedCombatant.hasBuff(e)) || event.hitType === HIT_TYPES.IMMUNE) {
+    const hitType = event.ability.type;
+    let checks = [];
+    let buffCheck = [];
+    let debuffCheck = [];
+    if (hitType === MAGIC_SCHOOLS.ids.PHYSICAL){
+      checks = this.checksPhysical;
+      buffCheck = [...this.buffCheckPhysical, ...this.buffCheckPhysAndMag];
+      debuffCheck = [...this.debuffCheckPhysical, ...this.debuffCheckPhysAndMag];
+    } else {
+      checks = this.checksMagical;
+      buffCheck = [...this.buffCheckMagical, ...this.buffCheckPhysAndMag];
+      debuffCheck = [...this.debuffCheckMagical, ...this.debuffCheckPhysAndMag];
+    }
+    if (checks.includes(spell) && !event.tick) {
+      debug && console.log(buffCheck);
+      debug && console.log(debuffCheck);
+      if (buffCheck.some((e) => this.selectedCombatant.hasBuff(e)) || event.hitType === HIT_TYPES.IMMUNE) {
```

Comment:
this function got a bit messy in the transition. idk if there is a great way to simplify it, though

Reply:
My own feeling is that it's fine for the time being. It's a little bit ugly, but like you said there's no obvious way to simplify things.